sdk/storage/Azure.Storage.Blobs/tests/ContainerClientTests.cs

```diff
@@ -403,7 +419,25 @@ public async Task CreateAsync_WithBlobServiceSas()
             // Arrange
             var containerName = GetNewContainerName();
             BlobServiceClient service = GetServiceClient_BlobServiceSas_Container(containerName);
-            BlobContainerClient container = InstrumentClient(service.GetBlobContainerClient(containerName));
+
+            BlobContainerSasPermissions permissions = BlobContainerSasPermissions.Read
+                | BlobContainerSasPermissions.Add
+                | BlobContainerSasPermissions.Create
+                | BlobContainerSasPermissions.Write
+                | BlobContainerSasPermissions.Delete
+                | BlobContainerSasPermissions.List;
+
+            BlobSasQueryParameters sasQueryParameters = GetContainerSas(
+                containerName: containerName,
+                permissions: permissions,
+                sasVersion: ToSasVersion(_serviceVersion));
+
+            BlobUriBuilder blobUriBuilder = new BlobUriBuilder(service.GetBlobContainerClient(containerName).Uri)
+            {
+                Sas = sasQueryParameters
+            };
+
+            BlobContainerClient container = new BlobContainerClient(blobUriBuilder.ToUri(), GetOptions());
```

Comment:
Nit - we need to instrument the client

Reply:
I'm pretty sure new XClient(..., GetOptions()) is the same as InstrumentClient(new XClient())

Reply:
No, they're different.  `GetOptions` will setup recording.  `InstrumentClient` will setup the sync/async and distributed tracing verification.